plugins/pur/core/src/main/java/org/pentaho/di/repository/pur/PurRepository.java

```diff
@@ -3060,7 +3417,13 @@ protected void saveTransOrJob( ISharedObjectsTransformer objectTransformer, Repo
     RepositoryFile file;
     if ( isUpdate ) {
       ObjectId id = element.getObjectId();
-      file = pur.getFileById( id.getId() );
+      readWriteLock.readLock().lock();
+      try {
+        file = pur.getFileById( id.getId() );
+      } finally {
+        readWriteLock.readLock().unlock();
```

Comment:
The Write lock should not be release between this and lower in the same method. The lock should be kept.